Packages/com.unity.inputsystem/InputSystem/Editor/Internal/SerializedPropertyHelpers.cs

```diff
@@ -0,0 +1,365 @@
+#if UNITY_EDITOR
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using UnityEditor;
+using UnityEngine.Experimental.Input.Utilities;
+
+namespace UnityEngine.Experimental.Input.Editor
+{
+    /// <summary>
+    /// Helpers for working with <see cref="SerializedProperty"/> in the editor.
+    /// </summary>
+    internal static class SerializedPropertyHelpers
```

Comment:
Can we have tests for these helpers?

Reply:
Hah, accidentally hit resolve...

TBH I generally think there isn't much of a point to those tests. Sometimes unit tests can be a great benefit but for the most part, I don't care much what an internal API somewhere deep in the system does. To me, this stuff here is equivalent to private methods somewhere.

What I want the tests to do is specify how the user facing parts (public API and UI) are to behave and ensure that's what they do. So for example here, `InputActionTreeView` is the user-facing piece of code (in this case, it's not public API but it is "public" UI) so that's where I want to test. `SerializedPropertyHelpers` and `InputActionSerializedHelpers` get extensively exercised by the tests for the tree view. Writing tests for them separately to me is largely equivalent to writing tests for some private method in `InputActionTreeView`.

We do have a few isolated unit tests for some stuff that proved tricky to get right but IMO wherever we can get away with not having them, we're usually better off.